src/hera/v1/task.py

```diff
@@ -163,6 +164,28 @@ def __rshift__(self, other: 'Task') -> 'Task':
         """
         return self.next(other)
 
+    def when(self, other: 'Task', operator: 'Operator', value: str):
```

Comment:
We can probably import the `Operator` here and use it as the type rather than use an annotation 🤔